src/net.h

```diff
@@ -1127,6 +1129,71 @@ class CConnman
     bool GetUseAddrmanOutgoing() const { return m_use_addrman_outgoing; };
     void SetNetworkActive(bool active);
     void OpenNetworkConnection(const CAddress& addrConnect, bool fCountFailure, CSemaphoreGrant&& grant_outbound, const char* strDest, ConnectionType conn_type, bool use_v2transport) EXCLUSIVE_LOCKS_REQUIRED(!m_unused_i2p_sessions_mutex);
+
+    /// Group of private broadcast related members.
+    class PrivateBroadcast
+    {
+    public:
+        /**
+         * Remember if we ever established at least one outbound connection to a
+         * Tor peer, including sending and receiving P2P messages. If this is
+         * true then the Tor proxy indeed works and is a proxy to the Tor network,
+         * not a misconfigured ordinary SOCKS5 proxy as -proxy or -onion. If that
+         * is the case, then we assume that connecting to an IPv4 or IPv6 address
+         * via that proxy will be done trough the Tor network and a Tor exit node.
+         */
+        std::atomic_bool m_outbound_tor_ok_at_least_once{false};
+
+        /**
+         * Check if private broadcast can be done to IPv4 or IPv6 peers and if so via which proxy.
+         * If private broadcast connections should not be opened to IPv4 or IPv6, then this will
+         * return an empty optional.
+         */
+        std::optional<Proxy> ProxyForIPv4or6() const;
+
+        /**
+         * Decide whether to open a private broadcast connection and if yes, to which network.
+         * We try to open a connection of this type only if there are transactions pending, but
+         * up to a limit.
+         * @param[in] prev_was_private_broadcast Whether the previous attempt to open a connection
+         * was an attempt to open a private broadcast connection (successful or not).
+         * @param[in] num_opened Number of private broadcast connections that are currently opened.
+         * @retval !std::nullopt The network to which to open the private broadcast connection.
+         * @retval std::nullopt Open any other connection (non-private broadcast).
+         */
+        std::optional<Network>
+        ShouldOpen(bool prev_was_private_broadcast, size_t num_opened) const;
+
+        /**
+         * Decrement by `n` the number of new connections of type `ConnectionType::PRIVATE_BROADCAST`
+         * to be opened by `CConnman::ThreadOpenConnections()`. Will not go negative, for example a
+         * value of 4 is ok to be decremented by 5 and will result in 0.
+         * @return the value preceding this operation
+         */
+        size_t NumToOpen() const;
```

Comment:
Fixed and added `@param[in]` comments.

What is enforced by clang (not by doxygen) via `-Wdocumentation` is a mismatch in the parameter names - e.g. if there is `@param[in] throughput ...` and the actual parameter is `int thrughput`, then it will give a warning (or error if `-Werror` is used).